File: packages/js-sdk/src/regex/detectors/in/aadhaar.ts
// Indian Aadhaar Number detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { inAadhaarChecksum } from '../../validators'

class InAadhaarDetector extends RegexDetector {
  entityType = EntityType.IN_AADHAAR
  score = 0.85
  contextKeywords = ['aadhaar', 'aadhar', 'uidai', 'unique identification']
  contextRequired = true

  // 12 digits starting with 2-9, with optional separators
  pattern = /\b[2-9]\d{3}[- ]?\d{4}[- ]?\d{4}\b/g
  validator = inAadhaarChecksum
}

registerRegion('in', InAadhaarDetector)

export { InAadhaarDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/in/aadhaar.py
"""Indian Aadhaar Number detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import in_aadhaar_checksum


@register_region("in")
class InAadhaarDetector(RegexDetector):
    entity_type = EntityType.IN_AADHAAR
    score = 0.85
    context_keywords = [
        "aadhaar", "aadhar", "uidai", "unique identification",
    ]
    context_required = True
    context_window = 50

    # 12 digits starting with 2-9, with optional separators
    pattern = re.compile(
        r"\b[2-9]\d{3}[- ]?\d{4}[- ]?\d{4}\b"
    )
    validator = staticmethod(in_aadhaar_checksum)
